cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -1762,9 +1768,47 @@ struct AntiExtrema<Decimal256> {
 };
 
 template <typename Type>
-struct GroupedMinMaxImpl : public GroupedAggregator {
+struct GroupedMinMaxTraits {
   using CType = typename TypeTraits<Type>::CType;
 
+  static void Update(CType* raw_mins, CType* raw_maxes, uint32_t g, CType val) {
+    raw_maxes[g] = std::max(raw_maxes[g], val);
+    raw_mins[g] = std::min(raw_mins[g], val);
+  }
+
+  static void Merge(CType* raw_mins, CType* raw_maxes, uint32_t g,
+                    const CType* other_raw_mins, const CType* other_raw_maxes,
+                    uint32_t other_g) {
+    raw_mins[g] = std::min(raw_mins[g], other_raw_mins[other_g]);
+    raw_maxes[g] = std::max(raw_maxes[g], other_raw_maxes[other_g]);
+  }
+};
```

Comment:
Note that this not very specific to min_max and you could simply have (assuming this is later useful for other types and aggregations?):
```c++
template <typename T>
struct GroupedValueTraits {
  using CType = typename TypeTraits<Type>::CType;

  static CType Get(const CType* values, uint32_t g) {
    return values[g];
  }
  static void Set(CType* values, uint32_t g, CType v) {
    values[g] = v;
  }
};

template <>
struct GroupedValueTraits<BooleanType> {
  static bool Get(const uint8_t* values, uint32_t g) {
    return BitUtil::GetBit(values, g);
  }
  static void Set(CType* values, uint32_t g, CType v) {
    BitUtil::SetBit(values, g, v);
  }
};
```